.gitlab-ci.yml

```diff
@@ -327,3 +328,94 @@ publish:docker-image:automatic:
   rules:
     - if: '$CI_COMMIT_BRANCH =~ /^(master|staging|production|feature-.+|[0-9]+\.[0-9]+\.([0-9]+|x))$/'
   extends: .template:publish:docker-image
+
+####################
+# Hardware testing #
+####################
+
+test:hardware:build:
+  <<: *convert_raspbian
+  variables:
+    RASPBERRYPI_PLATFORM: raspberrypi4
+  when: always
+  stage: build
+  script:
+    # Create the custom needed configurations
+    # Configure the ssh key permissions in the overlay
+    - chown -R root:root tests/ssh-public-key-overlay/root
+    - chmod 755 tests/ssh-public-key-overlay/root
+    - chmod 700 tests/ssh-public-key-overlay/root/.ssh
+    - chmod 600 tests/ssh-public-key-overlay/root/.ssh/authorized_keys
+    - ls
+    # Convert the image for the hardware tester
+    - env MENDER_ARTIFACT_NAME=${RASPBIAN_NAME}-mender
+      ./docker-mender-convert --disk-image input/${RASPBIAN_NAME}.img
+      --config configs/${RASPBERRYPI_PLATFORM}_config
+      --config configs/images/raspberrypi_raspbian_config
+      --config configs/testing/hardware-testing
+      --overlay tests/ssh-public-key-overlay
+    # Collect artifacts.
+    - mv deploy ${RASPBERRYPI_PLATFORM}
+    - tar czf ${RASPBERRYPI_PLATFORM}.tar.gz ${RASPBERRYPI_PLATFORM}
+    # Upload to temporary S3 bucket
+    - mender_ci_save_tmp_artifact ${RASPBERRYPI_PLATFORM}.tar.gz
+  artifacts:
+    paths:
+      - checksums
+
+test:hardware:flash:
```

Comment:
Please make both this and `test:hardware:acceptance` manual steps if you are going to merge this before going on PTO, it would be like deploying on a Friday. :-)

Reply:
Why? It won't run unless you start the first job

Reply:
But it depends on `test:hardware:build` which is set to "always"..?